src/sonic-yang-models/yang-models/sonic-port.yang

```diff
@@ -63,6 +63,44 @@ module sonic-port{
 					}
 				}
 
+				leaf autoneg {
+					description "Port auto negotiation mode";
+
+					type string {
+						pattern "on|off";
+					}
+				}
+				
+				leaf-list adv_speeds {
+					description "Port advertised speeds, valid value could be a list of interger or all";
+
+					type union {
+						type uint32 {
+							range 1..400000;
+						}
+						type string {
+							pattern "all";
+						}
+					}
+				}
+
+				leaf interface_type {
+					description "Port interface type";
+
+					type stypes:interface_type;
+				}
+
+				leaf-list adv_interface_types {
+					description "Port advertised interface type, valid value could be a list of stypes:interface_type or all";
+
+					type union {
+						type stypes:interface_type;
```

Comment:
This allows ["CR4", "all"] as well, if that is unwanted behavior then we should put some condition here.